### ISSUE TITLE
Can we add support for running generic Ruby scripts?

### ISSUE THREAD

**Issue by ioquatix** [rubygems/rubygems#6403]
https://github.com/rubygems/rubygems/blob/451ac56078a3deaacef640906070d3d8f7502570/lib/rubygems/ext/builder.rb#L122-L138

Can we add `/.rb/` or something that matches scripts ending in `.rb` and just run them with the Ruby interpreter? The current behaviour is inflexible.

**Comment by simi** [comment]
I have seen `extconf.rb` used to generate `Makefile` to just run custom script. Something like https://stackoverflow.com/a/38912965/319233.

What is your usecase @ioquatix?

**Comment by jchestershopify** [comment]
I am not a fan. Install scripts are a recurring cause of security issues, particularly credential harvesting.

**Comment by ioquatix** [comment]
@jchestershopify I agree with you, but according to the current interface, that is already possible. The only solution to that would be to use a sandbox for building native extensions. Which, I think is probably a good idea, but would be a totally separate issue.

Assuming that we just create a `makefile` which invokes a Ruby script, that's not practically different from what I'm proposing here, just that it's a simplification and it's more explicit. Otherwise, as @simi said, I can just use a hack to execute the code.

My use case is to use custom build tools, i.e. code which doesn't use any of the existing mechanisms.

**Comment by simi** [comment]
@ioquatix what kind of build tools? `cmake`, `make`, `cargo` are used to build native extensions. Is it possible to build native extension using `nix` or `npm`? I would like to prevent easy way of coupling multiple package managers together.
